receiver/kafkametricsreceiver/consumer_scraper.go

```diff
@@ -0,0 +1,179 @@
+	topicPartitions := make(map[string][]int32)
+	// currentOffset for each partition in matchedTopics
+	topicPartitionOffset := make(map[string]map[int32]int64)
+	for topic := range matchedTopics {
+		topicPartitions[topic] = make([]int32, 0)
+		topicPartitionOffset[topic] = make(map[int32]int64)
+		partitions, err := s.client.Partitions(topic)
+		if err != nil {
+			scrapeErrors.Add(err)
+			continue
+		}
+		for _, p := range partitions {
+			o, err := s.client.GetOffset(topic, p, sarama.OffsetNewest)
+			if err != nil {
+				scrapeErrors.Add(err)
+				continue
+			}
+			topicPartitions[topic] = append(topicPartitions[topic], p)
+			topicPartitionOffset[topic][p] = o
+		}
+	}
+	consumerGroups, listErr := s.clusterAdmin.DescribeConsumerGroups(matchedGrpIds)
+	if listErr != nil {
+		return metrics.ResourceMetrics(), listErr
+	}
+	for _, group := range consumerGroups {
+		grpMetrics := metrics.WithLabels(map[string]string{metadata.L.Group: group.GroupId})
+		grpMetrics.AddGaugeDataPoint(metadata.M.KafkaConsumerGroupMembers.Name(), int64(len(group.Members)))
+		groupOffsetFetchResponse, err := s.clusterAdmin.ListConsumerGroupOffsets(group.GroupId, topicPartitions)
+		if err != nil {
+			scrapeErrors.Add(err)
+			continue
+		}
+		for topic, partitions := range groupOffsetFetchResponse.Blocks {
+			// tracking matchedTopics consumed by this group
+			// by checking if any of the blocks has an offset
+			isConsumed := false
+			for _, block := range partitions {
+				if block.Offset != -1 {
+					isConsumed = true
+					break
+				}
+			}
+			grpTopicMetrics := grpMetrics.WithLabels(map[string]string{metadata.L.Topic: topic})
+			if isConsumed {
+				var lagSum int64 = 0
+				var offsetSum int64 = 0
+				for partition, block := range partitions {
+					grpPartitionMetrics := grpTopicMetrics.WithLabels(map[string]string{metadata.L.Partition: string(partition)})
+					consumerOffset := block.Offset
+					offsetSum += consumerOffset
+					grpPartitionMetrics.AddGaugeDataPoint(metadata.M.KafkaConsumerGroupOffset.Name(), consumerOffset)
+					// default -1 to indicate no lag measured.
+					var consumerLag int64 = -1
+					if partitionOffset, ok := topicPartitionOffset[topic][partition]; ok {
+						// only consider partitions with an offset
+						if block.Offset != -1 {
+							consumerLag = partitionOffset - consumerOffset
+							lagSum += consumerLag
+						}
+					}
+					grpPartitionMetrics.AddGaugeDataPoint(metadata.M.KafkaConsumerGroupLag.Name(), consumerLag)
+				}
+				grpTopicMetrics.AddGaugeDataPoint(metadata.M.KafkaConsumerGroupOffsetSum.Name(), offsetSum)
+				grpTopicMetrics.AddGaugeDataPoint(metadata.M.KafkaConsumerGroupLagSum.Name(), lagSum)
+			}
+		}
+	}
+
+	return metrics.ResourceMetrics(), scrapeErrors.Combine()
+}
+
+func createConsumerScraper(_ context.Context, config Config, saramaConfig *sarama.Config, logger *zap.Logger) (scraperhelper.ResourceMetricsScraper, error) {
+	groupFilter := regexp.MustCompile(config.GroupMatch)
```

Comment:
MustCompile will panic if it's an invalid regexp. Since it's a user-provided value that's not guaranteed to be valid I think you want to regexp.Compile and return the error.